File: 백준/Bronze/1159. 농구 경기/농구 경기.py
import sys
input = sys.stdin.readline

dic = dict()
state = False
result = []
for _ in range(int(input())):
    name = input().rstrip('\n')
    try:
        if dic[name[0]]: 
            dic[name[0]] += 1
        if dic[name[0]] == 5: 
            state = True
            result.append(name[0])
    except: dic[name[0]] = 1
if not state: print("PREDAJA")
else: 
    for i in sorted(result): print(i,end="")

File: 백준/Bronze/1159. 농구 경기/농구 경기.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        Map<String, Integer> map = new HashMap<>();
        int n = Integer.parseInt(br.readLine());
        Boolean state = false;
        ArrayList<String> li = new ArrayList<>();

        for (int i = 0; i < n; i++) {
            String name = String.valueOf(br.readLine().charAt(0));
            if (map.containsKey(name)) {
                map.put(name, map.get(name)+1);
                if (map.get(name) == 5) {
                    state = true;
                    li.add(name);
                }
            } else {
                map.put(name, 1);
            }
        }

        if (state) {
            Collections.sort(li);
            for (int i = 0; i < li.size(); i++) {
                System.out.print(li.get(i));
            }
        } else {
            System.out.println("PREDAJA");
        }
    }
}